andela/mervels-bn-backend
# #168131415 As a user, I can log into the Barefoot Nomad application via email & password

#### What does this PR do?
add user login using email and password
#### Description of Task to be completed?
- add user signing functionality
- add util class for handling responses
- add user validation when signing in 
- add test for signing feature 
- add swagger documentation
#### How should this be manually tested?
- clone the repo and checkout to the branch ` ft-login-user-168131415`
- npm install 
- install [redis-server](https://medium.com/@petehouston/install-and-config-redis-on-mac-os-x-via-homebrew-eb8df9a4f298) You can easily start it with `brew services start redis` after it has been installed for Mac users 
- on the terminal on working directory of the project you run `cp .env.example .env` to create an .env file. 
- populate the .env file with variables the default for `REDIS_SERVER=redis://localhost:6379` if not changed.
- run ` npm run test ` to check tests
- run `npm run start:dev` to start the server
- Go to postman and head over to `localhost:3000/api/v1/auth/signin`
- Set method name to `POST` and pass the payload 
```
{
	"userEmail": "wolly@stations.com",
	"userPassword": "123123"
}
```
- To run docker run the command `docker-compose up`
#### Any background context you want to provide?
- Redis server needs to be installed and started. 
- The user needs to be registered for them to be signed in.
#### What are the relevant pivotal tracker stories?
[#168131415](https://www.pivotaltracker.com/story/show/168131415)
#### Screenshots (if appropriate)
<img width="1088" alt="Screen Shot 2019-09-09 at 19 07 00" src="https://user-images.githubusercontent.com/18899285/64551317-0c06a400-d335-11e9-95b1-c00e804ac3bb.png">

#### Questions:
N/A
